xarray/core/formatting.py

```diff
@@ -419,20 +419,39 @@ def inline_index_repr(index, max_width=None):
     return repr_
 
 
-def summarize_index(name: Hashable, index, col_width: int, max_width: int = None):
+def summarize_index(
+    names: tuple[Hashable, ...], index, col_width: int, max_width: int | None = None
+) -> str:
     if max_width is None:
         max_width = OPTIONS["display_width"]
 
-    preformatted = pretty_print(f"    {name} ", col_width)
+    def prefix(index: int, length: int) -> str:
+        if index == 0 and length == 1:
+            return " "
+        elif index == 0:
+            return "┌"
+        elif index >= length - 1:
+            return "└"
+        else:
+            return "│"
 
-    index_width = max_width - len(preformatted)
+    preformatted = [
+        pretty_print(f"  {prefix(position, len(names))} {name}", col_width)
```

Comment:
Not sure if it is important, but this will call `__format__` on the names, in other places I _think_ we use `__str__`.
Maybe we should unify that at some point...